### PR TITLE
[CI] Build a Python wheel for aarch64 platform

### LINKED ISSUE #6180: Wheel support for linux aarch64
**Summary**
Installing xgboost on aarch64 via pip using command "pip3 install xgboost" tries to build wheel from source code.

**Problem description**
xgboost doesn't have wheel for aarch64 on PyPI repository. So, while installing xgboost via pip on aarch64, pip builds wheel for same resulting in it takes more time to install xgboost. Making wheel available for aarch64 will benefit aarch64 users by minimizing xgboost installation time.

**Expected Output**
Pip should be able to download xgboost wheel from PyPI repository rather than building it from source code.

@xgboost-team, please let me know if I can help you building wheel/uploading to PyPI repository. I am curious to make xgboost wheel available for aarch64. It will be a great opportunity for me to work with you.

### LINKED ISSUE #6243: Wheel for arm64
Reminder for following up on #6200 .

@hcho3 Should we target this for 1.3?

### PR DESCRIPTION
**Pull request by hcho3**
Closes #6243
Closes #6180

### CODE REVIEW

**Comment by hcho3** [comment]
@odidev Would you like to review?

**Comment by hcho3** [comment]
Closing to avoid spamming CI

**Comment by hcho3** [comment]
@trivialfis Can you review?

File: tests/ci_build/conda_env/macos_cpu_test.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
+name: cpu_test
```
Comment (trivialfis) [inline tests/ci_build/conda_env/macos_cpu_test.yml:1]: How's this different from `cpu_test.yml`

File: tests/ci_build/conda_env/aarch64_test.yml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+- distributed
+- hypothesis
+- graphviz
+- python-graphviz
+- codecov
+- cmake
+- ninja
+- boto3
+- jsonschema
+- numba
```
[the remark this answers is not in the document]
Comment (hcho3) [reply tests/ci_build/conda_env/aarch64_test.yml:25]: They are dependencies for `shap` package.

File: tests/ci_build/conda_env/macos_cpu_test.yml (hunk 1 above)
Comment (hcho3) [reply tests/ci_build/conda_env/macos_cpu_test.yml:1]: This one is missing numba and llvmlite.

File: tests/ci_build/conda_env/aarch64_test.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (trivialfis) [reply tests/ci_build/conda_env/aarch64_test.yml:25]: @hcho3 Thanks for the clarification.  Are you saying `shap` package has these dependencies but doesn't specify it in its package file, so we need to pull them explicitly as a downstream project?

[the remark this answers is not in the document]
Comment (hcho3) [reply tests/ci_build/conda_env/aarch64_test.yml:25]: @trivialfis The issue is that `pip install numba llvmlite`  fail on some platforms due to missing libraries etc.

[the remark this answers is not in the document]
Comment (trivialfis) [reply tests/ci_build/conda_env/aarch64_test.yml:25]: Thanks, good to know.  Am considering getting an arm board for fun.

**Review by trivialfis — APPROVED** [review]

**Comment by janaknat** [comment]
@hcho3 The xgboost wheel has an issue in CentOS. It's missing a library: libgomp.so.

Please see: https://github.com/geoffreyblake/arm64-python-wheel-tester/runs/1404000016?check_suite_focus=true

**Comment by hcho3** [comment]
@janaknat As the error message indicates, XGBoost requires the libgomp library. Please install libgomp with `sudo yum install libgomp`.

**Comment by janaknat** [comment]
Should the wheel not have all the libraries that are required by xgboost?

**Comment by hcho3** [comment]
@janaknat No, that's not reasonable expectation. `libgomp` is a major system library used by many applications, so it's best for system package manager (`yum`, `apt` etc) to manage it. Besides, Linux restricts the directories from which system libraries like `libgomp` can be loaded, unless `LD_LIBRARY_PATH` environment path is explicitly overridden by the user.

**Comment by geoffreyblake** [comment]
@hcho3, except a proper manylinux2014 wheel cannot dynamically link against that library as an implicit dependency according to [PEP 599's manylinux policy](https://www.python.org/dev/peps/pep-0599/#the-manylinux2014-policy).  It should be packaged in the wheel.   Most distros xgboost supports do not ship with libgomp.so by default as, Centos/RHEL for example do not.

**Comment by hcho3** [comment]
@geoffreyblake Thanks. I will run `auditwheel repair` to vendor libgomp. I've tested `auditwheel repair` on my machine, and it brings in a copy of libgomp.so into the wheel.

**Comment by hcho3** [comment]
@janaknat @geoffreyblake See #6461

**Pull request merged**